Reduce uncounted call arguments in Source/WebKit a bit more

annevk opened the pull request:
#### f1fe2e9f037e2a6076b549daea6da6d79b789cd1
<pre>
Reduce uncounted call arguments in Source/WebKit a bit more
<a href="https://bugs.webkit.org/show_bug.cgi?id=294749">https://bugs.webkit.org/show_bug.cgi?id=294749</a>

Reviewed by Chris Dumez.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/296459@main">https://commits.webkit.org/296459@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e967ff6fb0f977962358d459b12f2deb257a506

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108583 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28244 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18668 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113791 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58975 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28933 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36798 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82469 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111531 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22959 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97806 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62905 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22377 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58505 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92329 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15996 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116911 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35637 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26270 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91493 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36010 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94077 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91294 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23264 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36191 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13960 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/31455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35538 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41072 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35249 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38596 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36927 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebColorChooser.cpp:
```diff
@@ -39,21 +39,24 @@ using namespace WebCore;
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(WebColorChooser);
 
-WebColorChooser::WebColorChooser(WebPage* page, ColorChooserClient* client, const Color& initialColor)
```
We should probably just kept `page` and `client` as-is and used `page` and `client` in the constructor body instead of  `m_page` / `m_colorChooserClient`. There is no need to protect parameters so the static analysis wouldn't warn and the code would be safe.

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebColorChooser.cpp:
```diff
@@ -39,21 +39,24 @@ using namespace WebCore;
         return;
 
-    m_page->setActiveColorChooser(nullptr);
+    page->setActiveColorChooser(nullptr);
```
I think this would look nicer like so:
```
if (RefPtr page = m_page.get())
    page->setActiveColorChooser(nullptr);
```

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.cpp:
```diff
@@ -94,11 +94,16 @@ void WebContextMenuClient::stopSpeaking()
 void WebContextMenuClient::showContextMenu()
 {
-    m_page->contextMenu().show();
+    protectedPage()->protectedContextMenu()->show();
```
I think we may want to null check `m_page` instead of introducing `protectedPage()` here. Would look like:
```
if (RefPtr page = m_page.get())
    page->protectedContextMenu()->show();
```

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.cpp:
```diff
@@ -94,11 +94,16 @@ void WebContextMenuClient::stopSpeaking()
 
 #endif
 
+RefPtr<WebPage> WebContextMenuClient::protectedPage() const
```
I think we can remove.

annevk committed:
Reduce uncounted call arguments in Source/WebKit a bit more
https://bugs.webkit.org/show_bug.cgi?id=294749

Reviewed by Chris Dumez.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/296459@main
